### DISCUSSION TITLE
Error: `Fields "book" conflict because they have differing arguments. Use different aliases on the fields to fetch both if this was intentional.

### DISCUSSION THREAD

**Discussion by bappajubcse05** [yahoo/elide#2392] [Q&A] [answered]
I am trying to fetch all the records from a table/entity `book` and the books created by me using a graphQL query:

```
{  
  book(filter: "user ==Thomas") {
    edges {
      node {
        name        
      }
    }
  }
  book {
    edges{
      node{
        name
      }
    }
  }
}
```
But it is throwing below error:

```Fields "book" conflict because they have differing arguments. Use different aliases on the fields to fetch both if this was intentional.```

I am not sure how I can retrieve these details using a single query. If I need to use alias, how can I define the aliases in the query?

**Comment by aklish** [comment, marked as answer]
Rewrite your query more like:

```
{  
  mybooks1: book(filter: "user ==Thomas") {
    edges {
      node {
        name        
      }
    }
  }
  mybooks2: book {
    edges{
      node{
        name
      }
    }
  }
}
```

where mybooks1 and mybooks 2 are aliases you assign to each collection.

**Reply by Cardoso-topdev** [reply]
It works. Thank you for your answer.
